agent/os_helper.py

```diff
@@ -0,0 +1,269 @@
+import hashlib
+import os
+import platform
+import re
+from pathlib import Path
+from enum import Enum
+import pkg_resources
+
+import agent
+
+
+def detect_raspberry_pi():
+    metadata = {
+        'is_raspberry_pi': None,
+        'hardware_model': None,
+        'serial_number': None
+    }
+
+    proc_model = Path('/proc/device-tree/model')
+    proc_serial = Path('/proc/device-tree/serial-number')
+
+    if proc_model.is_file():
+        model = proc_model.open().read().strip('\0')
+        metadata['hardware_model'] = model
+        metadata['is_raspberry_pi'] = model.startswith('Raspberry Pi')
+
+    if proc_serial.is_file():
+        metadata['serial_number'] = proc_serial.open().read().strip('\0')
+
+    return metadata
+
+
+class Confinement(Enum):
+    NONE = 0
+    DOCKER = 1
+    BALENA = 2
+    SNAP = 3
+
+
+class Installation(Enum):
+    NONE = 0
+    DEB = 1
+    PYTHON_PACKAGE = 2
+    RPM = 3
+
+
+def detect_confinement():
+    if os.getenv('SNAP'):
+        return Confinement.SNAP
+    is_docker = 'docker' in open('/proc/1/cgroup', 'rt').read()
+    if is_docker:
+        if os.getenv('BALENA') or os.getenv('RESIN'):
+            return Confinement.BALENA
+        else:
+            return Confinement.DOCKER
+
+    return Confinement.NONE
+
+
+def detect_installation():
+    # For apt-based distros.
+    try:
+        import apt
+    except ModuleNotFoundError:
+        # For rpm-based distros.
+        try:
+            import rpm
+        except ModuleNotFoundError:
+            pass
+        else:
+            ts = rpm.ts()
+            mi = ts.dbMatch('name', 'wott-agent')
+            if mi.count() > 0:
+                package_header = mi.__next__()
+                if __file__.encode() in package_header[rpm.RPMTAG_FILENAMES]:
+                    return Installation.RPM
+    else:
+        cache = apt.Cache()
+        if __file__ in cache['wott-agent'].installed_files:
+            return Installation.DEB
+    # Other.
+    if isinstance(agent.__version__, pkg_resources.Distribution):
+        return Installation.PYTHON_PACKAGE
+    return Installation.NONE
+
+
+def get_packages():
+    # For apt-based distros.
+    try:
+        import apt
+    except ModuleNotFoundError:
```

Comment:
Instead of assuming that we're on an rpm-based system if we fail to import `apt` we should simply check if the system is a Debian or not (that's what we have `os_release` for). Otherwise, if for some reason `python-apt` is not installed on a Debian system this code will then try to import `rpm` which doesn't make any sense.